### PR TITLE
Add ROC_AUM loss 

### PR DESCRIPTION
**Pull request by cregouby**
Hello,

The ROC AUM loss is optimizing the ROC curve for unbalanced binary classification tasks.

This P.R. is a port of https://tdhock.github.io/blog/2025/mlr3torch-binary/

### CODE REVIEW

File: R/nnf-loss.R [hunk 1]
```diff
@@ -424,3 +424,63 @@ nnf_binary_cross_entropy_with_logits <- function(input, target, weight = NULL,
...
+#' Function that measures Area under the \eqn{Min(FPR, FNR)} (AUM)  between each
+#' element in the \eqn{input} and \eqn{target}. 
+#' 
+#' This is used for measuring the error of a binary reconstruction within highly unbalanced dataset, 
+#' where the goal is optimizing the ROC curve. 
+#'
+#' @param input Tensor of arbitrary shape
+#' @param target Tensor of the same shape as input. Should be the factor
+#' level of the binary outcome, i.e. with values `1L` and `2L`.
+#'
```
Comment (dfalbel) [inline R/nnf-loss.R:439]: Needs a description of the `roc_aum` parameter.

**Review by dfalbel — CHANGES_REQUESTED** [review]
Thanks @cregouby ! Just a minor comment.
